axum-macros/src/debug_handler.rs

```diff
@@ -284,6 +289,167 @@ fn check_inputs_impls_from_request(item_fn: &ItemFn, state_ty: Type) -> TokenStr
         .collect::<TokenStream>()
 }
 
+fn check_output_tuples(item_fn: &ItemFn) -> TokenStream {
+    let elems = match &item_fn.sig.output {
+        ReturnType::Type(_, ty) => match &**ty {
+            Type::Tuple(tuple) => &tuple.elems,
+            _ => return quote! {},
+        },
+        ReturnType::Default => return quote! {},
+    };
+
+    let handler_ident = &item_fn.sig.ident;
+
+    match elems.len() {
+        0 => quote! {},
+        n if n > 17 => syn::Error::new_spanned(
+            &item_fn.sig.output,
+            "Cannot return tuples with more 17 elements",
```

Comment:
```suggestion
            "Cannot return tuples with more than 17 elements",
```

Reply:
I was surprised by this too but we do actually allow 17 since the big T1..T16 impl as a final `R` at the end https://docs.rs/axum/latest/axum/response/trait.IntoResponse.html#impl-IntoResponse-for-(T1,+T2,+T3,+T4,+T5,+T6,+T7,+T8,+T9,+T10,+T11,+T12,+T13,+T14,+T15,+T16,+R)

Reply:
Right, edited the suggestion.